addon/mixins/ajax-request.js

```diff
@@ -39,11 +39,24 @@ const {
 } = Ember;
 const JSONAPIContentType = 'application/vnd.api+json';
 
+// Header names are case-insensitive
+// https://www.w3.org/Protocols/rfc2616/rfc2616.html
+function retrieveHeader(headers, name) {
+  var found = null;
+  Object.keys(headers).forEach((key) => {
+    let value = headers[key];
+    if (value.toLowerCase() === name.toLowerCase()) {
```

Comment:
At least one of the failing tests is due to this... We don't actually want to compare the value behind the key to the desired key, we want to match against the key.  I'm going to try to push a commit to this PR to get things passing.